pool/src/emissions/distributor.rs

```diff
@@ -399,6 +401,66 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_update_emission_no_overflow() {
+        let e = Env::default();
+        e.mock_all_auths();
+
+        let pool = testutils::create_pool(&e);
+        let samwise = Address::generate(&e);
+
+        e.ledger().set(LedgerInfo {
+            timestamp: 1501000000, // 10^6 seconds have passed
+            protocol_version: 22,
+            sequence_number: 123,
+            network_id: Default::default(),
+            base_reserve: 10,
+            min_temp_entry_ttl: 10,
+            min_persistent_entry_ttl: 10,
+            max_entry_ttl: 3110400,
+        });
+        // Supply of 1 trillion at 18 decimals
+        let supply: i128 = 1000000000000_000000000000000000;
+        let user_position: i128 = 1_000000000000000000;
+        e.as_contract(&pool, || {
+            let reserve_emission_data = ReserveEmissionData {
+                expiration: 1600000000,
+                eps: 0_01000000000000,
```

Comment:
can this test emit significantly more tokens?

the overflow will occur on `emissions * supply_scalar * SCALAR_7`. The this looks like it emits 10,000 BLND, which means the math will fit in i128 (intermediate math is ~1e36 i think).

If you emit like 1m BLND it should break the i128 math, and force it to scale to i256.